src/rcc.rs

```diff
@@ -111,8 +113,27 @@ impl AHB1 {
         // NOTE(unsafe) this proxy grants exclusive access to this register
         unsafe { &(*RCC::ptr()).ahb1rstr }
     }
+}
+
+/// AMBA High-performance Bus (AHB) peripheral registers
+pub struct AHB2 {
```

Comment:
Considering we're now implementing all AHB and APB busses one by one by basically writing out the same code again and again with changing numbers after the AHB/APB it might be worth turning these definitions and impls into a macro

Reply:
Agreed. However this doesn't belong in this PR, I will create an issue instead.